### PR TITLE
zilf 0.8 (new formula)

### PR DESCRIPTION
**Pull request by chkuendig**
- [✅] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [✅] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [✅] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [✅] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [❌] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?
-----

Today, the source code of many old Infocom Text Adventures (such as Zork) has been uploaded to http://github.com/historicalsource/ ([@textfiles Tweet](https://twitter.com/textfiles/status/1118002533908340736) for background). 

I thought it would be nice to have a toolchain available on homebrew to easily run these games from source. There's already [frotz](https://formulae.brew.sh/formula/frotz) available on homebrew to play the compiled games, but to compile them, you'd have to download zilf manually.

_I wasn't able to get zilf to compile the samples in any other directory then "bin/" (just unzipping it locally and renaming the folder means some includes start to break), but other than that, this passes all audits._ **Update: This is fixed now**

edit: there was a missing space in the file. It's corrected now. 

PS: If you want to run zork:
```
git clone git://github.com/historicalsource/zork1.git
cd zork1
zilf zork1.zil
zapf zork1.zap
frotz zork1.z3
```
and you should get:
```
ZORK I: The Great Underground Empire
Infocom interactive fiction - a fantasy story
Copyright (c) 1981, 1982, 1983, 1984, 1985, 1986 Infocom, Inc. All rights reserved.
ZORK is a registered trademark of Infocom, Inc.
Release 0 / Serial number 190416

West of House
You are standing in an open field west of a white house, with a boarded front door.
There is a small mailbox here.

>
```

### CODE REVIEW

**Comment by fxcoudert** [comment]
```
  * Non-executables were installed to "/usr/local/opt/zilf/bin"
    The offending files are:
      /usr/local/opt/zilf/bin/Antlr3.Runtime.dll
      /usr/local/opt/zilf/bin/Zilf.Common.dll
      /usr/local/opt/zilf/bin/zilf.exe
      /usr/local/opt/zilf/bin/Zilf.Emit.dll
      /usr/local/opt/zilf/bin/zapf.exe
```

File: Formula/zilf.rb [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+  depends_on "mono"
+
+  def install
+    prefix.install "README.txt"
+    bin.install Dir["bin/*.dll", "bin/*.exe"]
+    prefix.install Dir["doc", "library", "sample"]
+    (bin/"zilf").write <<~EOS
+      #!/bin/sh
+      mono #{bin}/zilf.exe "$@"
```
Comment (taradinoc) [inline Formula/zilf.rb]: >I wasn't able to get zilf to compile the samples in any other directory then "bin/" (just unzipping it locally and renaming the folder means some includes start to break)

You can pass the include path explicitly with `-ip`:

```suggestion
      mono #{bin}/zilf.exe -ip "#{prefix}/library" "$@"
```

**Comment by chkuendig** [comment]
updated the formula, thanks for the feedback

**Comment by chkuendig** [comment]
Build now passes `brew audit --strict <formula>` - anything else I can help with to get this merged?

**Comment by zbeekman** [comment]
6 watchers and 2 forks on bitbucket.

I have mixed feelings about merging this without it reaching higher popularity.

On the one hand, not needing to bottle it is nice, but on the other hand the formula has to basically manually install each components since there's no native build/install script for macOS provided upstream.

I'm inclined to say this is a "won't merge" on popularity grounds, but others can weigh in.

**Comment by SMillerDev** [comment]
We have an audit check for popularity. If it doesn't trigger it's good to go for me.

**Comment by zbeekman** [comment]
FYI, I think the audit check only works on certain sources (GitHub, maybe
others, NOT BitBucket, AFAICT).

**Comment by SMillerDev** [comment]
Hi, this seems like a nice software, but not notable enough for us to package in Homebrew for now. This is not a decision on the merits of the software, but we cannot distribute every single piece of software and have to draw a line on “notability” somewhere. Of course, we would be happy to reconsider in the future.

Please consider hosting it in a personal tap, which is very easy to do: https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Pull request closed without merging**